src/containers/shared/components/SearchResults/SearchResults.tsx

```diff
@@ -0,0 +1,291 @@
+            </div>
+            <div className="search-result-metric-chip">
+              HOLDERS: {localizeNumber(resultContent.metrics.holders)}
+            </div>
+            <div className="search-result-metric-chip">
+              <div>
+                TRUSTLINES: {localizeNumber(resultContent.metrics.trustlines)}
+              </div>
+            </div>
+          </div>
+          <div className="search-result-row-line-two">
+            <div>Issuer:</div>
+            {resultContent.issuer && (
+              <Link
+                to={`/accounts/${resultContent.issuer}`}
+                onClick={onClick}
+                className="issuer-link"
+              >
+                <div>
+                  {resultContent.meta.issuer.name
+                    ? `${resultContent.meta.issuer.name} (${resultContent.issuer})`
+                    : resultContent.issuer}
+                </div>
+              </Link>
+            )}
+          </div>
+          <div className="search-result-row-line-three">
+            {resultContent.meta.issuer.domain && (
+              <>
+                <div>Website:</div>
+                <div>
+                  <Link
+                    to={`https://${resultContent.meta.issuer.domain}/`}
+                    className="issuer-link"
+                    target="_blank"
+                    rel="noreferrer"
+                  >
+                    {parseDomain(resultContent.meta.issuer.domain)}
+                  </Link>
+                </div>
+              </>
+            )}
+          </div>
+        </div>
+      </div>
+    </Link>
+  )
+}
+
+const SearchResults = ({
+  currentSearchValue,
+  setCurrentSearchInput,
+}: SearchResultsProps): JSX.Element => {
+  const xrplmetaWSRef = useRef<WebSocket | null>(null)
+  const [rawCurrentSearchResults, setRawCurrentSearchResults] = useState<any[]>(
+    [],
+  )
+  const [XRPUSDPrice, setXRPUSDPrice] = useState(0.0)
+
+  const connect = () => {
+    xrplmetaWSRef.current = new WebSocket('wss://s1.xrplmeta.org', 'tokens')
+
+    xrplmetaWSRef.current.onmessage = (event) => {
+      const results = JSON.parse(event.data).result.tokens
+
+      const filteredResults = results.filter(
+        (result) =>
+          result.metrics.trustlines > 50 &&
+          result.metrics.holders > 50 &&
+          result.metrics.marketcap > 0 &&
+          result.metrics.volume_7d > 0,
+      )
+
+      setRawCurrentSearchResults(filteredResults)
+    }
+
+    xrplmetaWSRef.current.onclose = () => {
+      attemptReconnect()
+    }
+
+    xrplmetaWSRef.current.onerror = () => {
+      xrplmetaWSRef.current?.close()
+      attemptReconnect()
+    }
+  }
+
+  const attemptReconnect = () => {
+    // Exponential backoff logic
+    setTimeout(() => {
+      connect()
+    }, 500)
+  }
+
+  // xrpl cluster temporary socket connection for XRP/USD conversion rate
+  useEffect(() => {
+    connect()
+    const xrplClusterSocket = new WebSocket('wss://xrplcluster.com')
```

Comment:
this should use existing socket connection all other pages use.

Reply:
This one needs to be specific to xrplcluster from XRPLF as opposed to the rippled node we use in the rest of the app in order to run the psuedo price oracle command that retrieves the XRP/USD conversion rate. This was only offered through the websocket from my understanding, which is why I establish and then immediately close it after a single command too, and should be replaced as soon as actual oracles goes live

Reply:
It can be retrieved via any rippled/clio node.  The account is what makes it serve as an oracle.